### PR TITLE
py-nanobind: add v2.1.0 and v2.2.0

### PR DESCRIPTION
**Pull request by garth-wells**
Updates dependencies based on the `pyproject.toml` contents.

~~Requires #46883~~

### CODE REVIEW

**Review by wdconinc — APPROVED** [review]
Confirmed commit hashes. Builds correctly for both new versions. Python can `import nanobind` and print a version.

**Comment by wdconinc** [comment]
Deferring to maintainer before merging.

**Review by chrisrichardson — APPROVED** [review]

**Pull request merged**
